File: v1/src/test/java/com/api/v1/payment/PaymentDeletionByOrderNumberTest.java
package com.api.v1.payment;

import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.reactive.server.WebTestClient;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class PaymentDeletionByOrderNumberTest {

    @Autowired
    WebTestClient webTestClient;

    @Test
    @Order(1)
    void testSuccessfulPaymentDeletion() {
        webTestClient
                .delete()
                .uri("api/v1/payments/%s".formatted("20240001"))
                .exchange()
                .expectStatus()
                .is2xxSuccessful();
    }

    @Test
    void testUnSuccessfulPaymentDeletion() {
        webTestClient
                .delete()
                .uri("api/v1/payments/%s".formatted("20240002"))
                .exchange()
                .expectStatus()
                .is2xxSuccessful();
    }

}

File: v2/src/test/kotlin/com/api/v2/payment/PaymentDeletionByOrderNumberTest.kt
package com.api.v2.payment

import org.junit.jupiter.api.MethodOrderer
import org.junit.jupiter.api.Order
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestMethodOrder
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.web.reactive.server.WebTestClient

@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class PaymentDeletionByOrderNumberTest {

    @Autowired
    lateinit var webTestClient: WebTestClient

    @Test
    @Order(1)
    fun testSuccessfulAllDeletion() {
        webTestClient
            .delete()
            .uri("api/v1/payments/${20240001}")
            .exchange()
            .expectStatus()
            .is2xxSuccessful()
    }

    @Test
    fun testUnsuccessfulAllDeletion() {
        webTestClient
            .delete()
            .uri("api/v1/payments/${20240001}")
            .exchange()
            .expectStatus()
            .is5xxServerError()
    }

}